### PR TITLE
feat: support aten.arange.start_step dynamo converter

### LINKED ISSUE #2492: `aten.arange.start_step`

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

Support `aten.arange.start_step` dynamo converter.

Fixes #2492 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by apbose — COMMENTED** [review]
Can this be implemented in the evaluator instead?

File: py/torch_tensorrt/dynamo/conversion/impl/sequence.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+def arange(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    start: Union[int, float],
+    end: Union[int, float],
+    step: Union[int, float] = 1,
+) -> TRTTensor:
+    values = np.arange(start, end, step)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/sequence.py:20]: Consider switching output to `np.array` type + making evaluator

File: py/torch_tensorrt/dynamo/conversion/ops_evaluators.py [hunk 2]
```diff
@@ -23,7 +24,9 @@ def getitem_validator(getitem_node: Node) -> bool:
 
 # TODO: Subsequent evaluators should be registered here with their own validators
 @dynamo_tensorrt_converter(operator.getitem, capability_validator=getitem_validator)
+@dynamo_tensorrt_converter(builtins.getattr)
 @dynamo_tensorrt_converter(torch.ops.aten.detach.default)
+@dynamo_tensorrt_converter(torch.ops.aten.arange.start_step)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/ops_evaluators.py:29]: This may not work out of the box in the `torch.compile` workflow, since it will invoke `torch.ops.aten.arange.start_step()`, which can create `FakeTensor` objects. A temporary workaround is to add a new converter in this file just for `arange`.

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/ops_evaluators.py]: @gs-olive Thanks for the explanation! I fixed it with `np.arange`

File: py/torch_tensorrt/dynamo/conversion/ops_evaluators.py [hunk 3]
```diff
@@ -23,6 +25,7 @@ def getitem_validator(getitem_node: Node) -> bool:
 
 # TODO: Subsequent evaluators should be registered here with their own validators
 @dynamo_tensorrt_converter(operator.getitem, capability_validator=getitem_validator)
+@dynamo_tensorrt_converter(builtins.getattr)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/ops_evaluators.py:28]: What does the `getattr` builtin do in the FX graph? Are there any sample uses of it in models? It may need a validator like `operator.getitem`

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/ops_evaluators.py:28]: Thanks for the comment! Yes, `builtins.getattr` is used in the test here: 
```python
class Arange(nn.Module):
    def forward(self, x):
        return torch.ops.aten.arange.start_step(start, x.shape[0], step)
```
where `x.shape` calls `builtins.getattr`, as the graph below: 
```
.INFO:harness:FX graph= graph():
    %x : [num_users=1] = placeholder[target=x]
    %getattr_1 : [num_users=1] = call_function[target=builtins.getattr](args = (%x, shape), kwargs = {})
    %getitem : [num_users=1] = call_function[target=operator.getitem](args = (%getattr_1, 0), kwargs = {})
    %arange_start_step : [num_users=1] = call_function[target=torch.ops.aten.arange.start_step](args = (3, %getitem, 3), kwargs = {})
    return arange_start_step
```

I thought about adding `validator`, but it depends on the attributes that the caller has. It seems not general to only allow `shape` attr I think? Do you have any idea about validator?

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/ops_evaluators.py:28]: I don't think `builtins.getattr` is a standard op, and should not appear in Dynamo graphs after ATen tracing. Could you try running the test case with `use_dynamo_tracer=True`?

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/ops_evaluators.py:28]: Cool, it disappeared, thanks!

File: py/torch_tensorrt/dynamo/conversion/ops_evaluators.py [hunk 4]
```diff
@@ -35,3 +36,14 @@ def generic_evaluator(
...
+
+@dynamo_tensorrt_converter(torch.ops.aten.arange.start_step)
+def aten_ops_arange_start_step(
+    ctx: ConversionContext,
+    target: Target,
+    args: Tuple[Argument, ...],
+    kwargs: Dict[str, Argument],
+    name: str,
+) -> Union[TRTTensor, Sequence[TRTTensor]]:
+    return np.arange(*args)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/ops_evaluators.py:49]: A very minor comment - using `np.arange(*args)` requires the schemas of `arange.start_step` and `np.arange` to be the same position-wise. In this case they are the same and are unlikely to change, so I think this is fine, just wanted to mention this scheme may not hold between Torch and Numpy in general.

**Review by gs-olive — APPROVED** [review]
Looks good to me. Made a small comment with no action required.
